### DISCUSSION TITLE
Code block copy removes leading tabs on first line

### DISCUSSION THREAD

**Discussion by KiGamji** [danny-avila/LibreChat#5075] [Feature Requests & Suggestions]
### What happened?

When a code block with leading tabs is generated and displayed in the UI, the "Copy code" button does not preserve the tabs on the first line when the code is copied to the clipboard. This causes the pasted code to lose the intended indentation, and differs from how the markdown code block was created.

### Steps to Reproduce

1. Generate a code block with tabbed indents using markdown, ensuring the first line begins with a tab.
2. Click the "Copy code" button.
3. Paste the code and check it out.

### Screenshots

![image](https://github.com/user-attachments/assets/4661b25f-946d-4174-81e2-772209c1c14f)

![image](https://github.com/user-attachments/assets/b4a25bcb-5a74-4ef1-88f7-681e4edcf05e)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
This is intended, the text is trimmed. Otherwise there might be trailing new lines

**Reply by KiGamji** [reply]
hmm, I understand
I have opened this issue, because this behavior introduces a significant usability issue when working with the model and editing a huge code block, where indentation is essential

**Reply by danny-avila** [reply]
It should be a simple fix. Could just remove new lines but not tabs at the start or end. PRs welcome, it should be one file change

**Reply by danny-avila** [reply]
@berry-13
